plugins/kafka/src/api/types.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createApiRef } from '@backstage/core';
+
+export const kafkaApiRef = createApiRef<KafkaApi>({
+  id: 'plugin.kafka.service',
+  description:
+    'Used by the Kafka plugin to make requests to accompanying backend',
+});
+
+export type ConsumerGroupOffsetsResponse = {
+  consumerId: string;
+  offsets: {
+    topic: string;
+    partitionId: number;
+    topicOffset: string;
+    groupOffset: string;
+  }[];
+};
+
+export interface KafkaApi {
+  getConsumerGroupOffsets(
+    consumerGroup: string,
```

Comment:
What do you think about consuming an entity here instead, and reading the annotation within the implementation? That way we're a bit better set up for additions to the model to e.g. select which cluster to talk to

Reply:
Interesting, but I think the problem is that this will cause coupling of the parsing of the entity into the API with the backend (which might be used by other places in the plugin in the future that may not have the same exact entity). I saw that other plugins do this (for example the Kubernetes or CircleCI plugins which I used for inspiration).

Reply:
You're thinking that it would be a hassle to change the REST API as it would require you to update both plugins at once? I'm thinking you'd either make the changes backwards compatible or add a new API version, although we could also grab the entity in the backend, and let the parsing happen there.

Tbh we're looking at doing a similar refactor of the kubernetes plugin, where we pass the entity rather than a service ID. Same reason there, we want the entity to be able to be involved in the cluster and resource selection. The idea is that it provides a lot more flexibility for custom implementations of the resource selection and makes the entire plugin a lot more reusable.

Another aspect is that we're looking way to add access control, and that's primarily based around entities and ownership of them. It would ofc be a lot easier to add access control in the Kafka plugin backend if you have the entire entity from the catalog to work with rather than just the `consumerGroup`. This is ofc not too relevant right now as we're far away from that, but your reason for chosing this design is exactly why I'd like to reconsider it, as a lot of plugins base their design on existing plugins.

I think the main argument against passing an entity is that it doesn't make the plugin as usable from outside of entity contexts, but tbh the plugin components already add that dependency so not sure that's useful ^_>

Reply:
I meant that I imagine there will be other views that show you consumer offsets. For example, imagine we also add a general "kafka cluster" page that just shows you information about the cluster, then you can click on a topic and see all consumers, and click on a consumer and see its offsets. So in that case you'll just have the consumer ID and you'd want to see its offsets. Does it make sense to build an entity and send it to server just for that? 

Also I see the backend API having more APIs that aren't related to a specific component - like getting all topics, looking into message types in a topic, etc. These might not have an entity behind them necessarily.

Reply:
Yeah that's another reason to stick to not including entities. Tbh in that case I'd imagine the entity pages and the global browsing pages to be backed by two different APIs, which ofc can mostly share implementation in the backend. 

Another option, which I think would be better in that case, is to model kafka topics/clusters as `Resource` entities, which would ofc also be consumable by the same endpoint.

It's all very undecided and needs more thinking though, so happy to not block this PR on that kind of refactor and design work.

Reply:
Yes I think prefer to refactor this later, maybe once more API endpoints are in use here.